src/edges.cpp

```diff
@@ -1,8 +1,55 @@
 #include "edges.h"
+#include <set>
 
 Eigen::MatrixXi edges(const Eigen::MatrixXi &F)
 {
-  Eigen::MatrixXi E;
-  // ADD YOUR CODE HERE
-  return E;
+    // O(|F|log|F|)
+	// Use additional O(|F|) storage to create edges set. 
+    // std::set supports O(log(|F|)) insert and lookup. 
+	// NOTE: Can change to unordered_set to support O(1) lookup if I provide hash function for pair<int,int>
+	Eigen::MatrixXi E;
+	Eigen::MatrixXi EBig(3*F.rows(), 2);
+	std::set<std::pair<int, int>> m;
+	int lostRows = 0;
+	std::pair<int, bool> p;
```

Comment:
Seems like this has a typo. The code will work if this line is replaced with

    std::pair<int, int> p;